content/en/docs/concepts/workloads/controllers/deployment.md

```diff
@@ -311,8 +311,7 @@ up to 3 replicas, as well as scaling down the old ReplicaSet to 0 replicas.
 ### Rollover (aka multiple updates in-flight)
 
 Each time a new Deployment is observed by the Deployment controller, a ReplicaSet is created to bring up
-the desired Pods. If the Deployment is updated, the existing ReplicaSet that controls Pods whose labels
-match `.spec.selector` but whose template does not match `.spec.template` are scaled down. Eventually, the new
+the desired Pods. If the Deployment is updated, the existing ReplicaSet with its label(s) in `.spec.selector` matching the pod's template label(s) in `.spec.template` are scaled up but those pod's template label(s) in `.spec.template` not matching with the ReplicaSet's labels in `.spec.selector` are scaled down. Eventually, the new
```

Comment:
Hi @MIhirMishra . Thanks for the updates.
- Minor point: You could use the word labels instead of "label(s)"
- The big sentence could be split into two parts. Does it make sense to clarify the second part of the sentence
  by adding ReplicaSets  .... are scaled down.

Reply:
Being a non native english speaker - my brain alway think of keeping multiple labels when my eyes see labels; it was more of how formally I still read english. Label(s) communicate that it could take either one or multiple labels - but that is just me.